Undefined references with libPocoCrypto shared library

I built the framework using [the continuumio miniconda docker image build script](https://github.com/mantidproject/conda-recipes/blob/master/docker/framework/run_docker_build_continuumio-mc2.sh) and ran into the error of undefined references related to libPocoCryto.so library:

```
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: warning: libssl.so.1.0.0, needed by /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoNetSSL.so, not found (try using -rpath or -rpath-link)
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: warning: libcrypto.so.1.0.0, needed by /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoNetSSL.so, not found (try using -rpath or -rpath-link)
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoCrypto.so: undefined reference to `EVP_get_digestbyname'
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoCrypto.so: undefined reference to `BIO_s_file'
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoCrypto.so: undefined reference to `RSA_size'
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoCrypto.so: undefined reference to `BN_new'
/build_artifacts/mantid-framework_1551269221778/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /build_artifacts/mantid-framework_1551269221778/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/libPocoNetSSL.so: undefined reference to `ERR_error_string_n'
.
.
.
```

Reading, it seems related to OpenSSL version 1.1.1 vs. 1.0.2

--- a/framework/conda_build_config.yaml
+++ b/framework/conda_build_config.yaml
@@ -53,6 +53,8 @@ tbb:
   - 2018_20170919
 librdkafka:
   - 0.11
+openssl:
+  - 1.0.2
 
 zip_keys:
   - python
@@ -87,6 +89,8 @@ pin_run_as_build:
   tbb:
     max_pin: x.x
   librdkafka:
+    max_pin: x.x
+  openssl:
     max_pin: x.x
 #  muparser:
 #    max_pin: x.x

--- a/framework/meta.yaml
+++ b/framework/meta.yaml
@@ -30,7 +30,7 @@ requirements:
     - tbb       {{ tbb }}
     - librdkafka  {{ librdkafka }}
     - readline
-    - openssl
+    - openssl   {{ openssl }}
     - python-dateutil
     - h5py
     - mpi4py

